automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala

```diff
@@ -16,18 +16,24 @@ class DataUseAwareSearchSpec extends FreeSpec with WebBrowserSpec with UserFixtu
 
   // We are only testing UI mechanics because the business logic of RP matching is extensively tested lower in the stack.
   "Data Library" - {
-    "Open and close the RP modal" in withWebDriver { implicit driver =>
+    "Repeatedly open and close the RP modal" in withWebDriver { implicit driver =>
       val user = UserPool.chooseAnyUser
 
       withSignIn(user) { _ =>
         val page = new DataLibraryPage().open
 
-        page.openResearchPurposeModal()
+        page.isShowingResearchPurposeModal shouldBe false
 
+        page.openResearchPurposeModal()
         page.isShowingResearchPurposeModal shouldBe true
 
         page.dismissResearchPurposeModal()
+        page.isShowingResearchPurposeModal shouldBe false
 
+        page.openResearchPurposeModal()
+        page.isShowingResearchPurposeModal shouldBe true
+
+        page.dismissResearchPurposeModal()
```

Comment:
It is _now_... but the point of this check is to make sure we know if it ever decides not to

Reply:
Maybe we should have a utility for checking modals for closeability. I don't know exactly how it would work, but it would be great to have some `modal.assertWorks()` that we can drop in to any test that involves modals